### PR TITLE
Add custom components docs section.

### PR DESCRIPTION
**Pull request by Hyperkid123**
### Changes
- add custom components demo section to docs

cc @jinxac

### CODE REVIEW

File: packages/react-renderer-demo/src/examples/components/custom-components/sample-example.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      component: 'custom-component',
+      name: 'some-name',
+      label: 'Custom component prop',
+      validate: [{ type: validatorTypes.REQUIRED, message: 'I am a required field' }]
+    }
+  ]
+};
+
+const SampleExample = () => {
+  return (
```
Comment (rvsia) [inline packages/react-renderer-demo/src/examples/components/custom-components/sample-example.js:33]: We don't return block here

File: packages/react-renderer-demo/src/pages/custom-examples/sample-example.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+# Sample example
+
+## Use markdown syntax to write text
+
+## What to include?
+- description of the component
+- what issue it can help solve
+- include component preview in a form
```
Comment (rvsia) [inline packages/react-renderer-demo/src/pages/custom-examples/sample-example.md:14]: Maybe I would use this page to say: 

"Hey if you need custom component, use useField and check this example"

We can put the text you used to `development setup` page. :thinking:

File: packages/react-renderer-demo/src/components/navigation/schemas/schema.js [hunk 3]
```diff
@@ -57,6 +58,12 @@ const schema = [
     linkText: 'Optimization',
     link: 'optimization'
   },
+  {
+    title: 'Custom components demo',
```
Comment (rvsia) [inline packages/react-renderer-demo/src/components/navigation/schemas/schema.js:62]: what about to remove `demo`?

**Review by rvsia — COMMENTED** [review]
Just some suggestions.

One thing to do: after this is merged, update DocSearch config, there are defined entrypoints and we need to add each new menu subsection as an entrypoint to let algolia explore the menu.

File: packages/react-renderer-demo/src/components/navigation/schemas/schema.js (hunk 3 above)
Comment (Hyperkid123) [reply packages/react-renderer-demo/src/components/navigation/schemas/schema.js:62]: We already have custom components entry in the docs. I would like users to find the guide how to create the custom components before looking at this.

File: packages/react-renderer-demo/src/pages/custom-examples/sample-example.md (hunk 2 above)
Comment (Hyperkid123) [reply packages/react-renderer-demo/src/pages/custom-examples/sample-example.md:14]: I fully expect this example to get deleted. This is here just for @jinxac so he knows where to pud his demo.

File: packages/react-renderer-demo/src/components/navigation/schemas/schema.js (hunk 3 above)
Comment (rvsia) [reply packages/react-renderer-demo/src/components/navigation/schemas/schema.js:62]: hmm so maybe 'Custom components examples'?

Comment (Hyperkid123) [reply packages/react-renderer-demo/src/components/navigation/schemas/schema.js:62]: I think examples might be enough.

**Review by rvsia — APPROVED** [review]

**Pull request merged**